<a href="https://colab.research.google.com/github/wahyunh10/Project-Ecommerce-Shipping-Clasification-Modeling/blob/main/Stage_1_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Data Preprocessing**

Import the csv file as panda DataFrames and examine it.

In [1]:
# import library

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df = pd.read_csv('Train.csv')
df

ID Warehouse_block Mode_of_Shipment  Customer_care_calls  \
0          1               D           Flight                    4   
1          2               F           Flight                    4   
2          3               A           Flight                    2   
3          4               B           Flight                    3   
4          5               C           Flight                    2   
...      ...             ...              ...                  ...   
10994  10995               A             Ship                    4   
10995  10996               B             Ship                    4   
10996  10997               C             Ship                    5   
10997  10998               F             Ship                    5   
10998  10999               D             Ship                    2   

       Customer_rating  Cost_of_the_Product  Prior_purchases  \
0                    2                  177                3   
1                    5                  216                2   
2                    2                  183                4   
3                    3                  176                4   
4                    2                  184                3   
...                ...                  ...              ...   
10994                1                  252                5   
10995                1                  232                5   
10996                4                  242                5   
10997                2                  223                6   
10998                5                  155                5   

      Product_importance Gender  Discount_offered  Weight_in_gms  \
0                    low      F                44           1233   
1                    low      M                59           3088   
2                    low      M                48           3374   
3                 medium      M                10           1177   
4                 medium      F                46           2484   
...                  ...    ...               ...            ...   
10994             medium      F                 1           1538   
10995             medium      F                 6           1247   
10996                low      F                 4           1155   
10997             medium      M                 2           1210   
10998                low      F                 6           1639   

       Reached.on.Time_Y.N  
0                        1  
1                        1  
2                        1  
3                        1  
4                        1  
...                    ...  
10994                    1  
10995                    0  
10996                    0  
10997                    0  
10998                    0  

[10999 rows x 12 columns]

In [3]:
df.shape

(10999, 12)

**Data Exploration**

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10999 entries, 0 to 10998
Data columns (total 12 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   ID                   10999 non-null  int64 
 1   Warehouse_block      10999 non-null  object
 2   Mode_of_Shipment     10999 non-null  object
 3   Customer_care_calls  10999 non-null  int64 
 4   Customer_rating      10999 non-null  int64 
 5   Cost_of_the_Product  10999 non-null  int64 
 6   Prior_purchases      10999 non-null  int64 
 7   Product_importance   10999 non-null  object
 8   Gender               10999 non-null  object
 9   Discount_offered     10999 non-null  int64 
 10  Weight_in_gms        10999 non-null  int64 
 11  Reached.on.Time_Y.N  10999 non-null  int64 
dtypes: int64(8), object(4)
memory usage: 1.0+ MB


Observation
1. Data consists of 10999 rows
2. There is no data that is null or missing value
3. There doesn't seem to be a glaring issue with the data types for each column (it's appropriate)

In [5]:
df.sample(5)

ID Warehouse_block Mode_of_Shipment  Customer_care_calls  \
3143    3144               F             Road                    3   
9552    9553               D             Ship                    6   
4431    4432               B             Ship                    4   
10584  10585               D             Ship                    5   
10111  10112               F             Road                    4   

       Customer_rating  Cost_of_the_Product  Prior_purchases  \
3143                 3                  253                3   
9552                 1                  276                2   
4431                 2                  257                3   
10584                5                  187                3   
10111                4                  258                3   

      Product_importance Gender  Discount_offered  Weight_in_gms  \
3143              medium      M                 7           5085   
9552                 low      M                10           5108   
4431              medium      M                 8           5045   
10584             medium      F                 5           4613   
10111               high      M                 2           5420   

       Reached.on.Time_Y.N  
3143                     0  
9552                     0  
4431                     1  
10584                    0  
10111                    0

**Exploratory Data Analysis**

In [6]:
# grouping of columns by type
nums = ['Customer_care_calls', 'Customer_rating', 'Prior_purchases', 'Discount_offered', 'Cost_of_the_Product', 'Weight_in_gms', 'Reached.on.Time_Y.N']
cats = ['Mode_of_Shipment', 'Product_importance', 'Gender','Warehouse_block']

In [7]:
df[nums].describe()

Customer_care_calls  Customer_rating  Prior_purchases  \
count         10999.000000     10999.000000     10999.000000   
mean              4.054459         2.990545         3.567597   
std               1.141490         1.413603         1.522860   
min               2.000000         1.000000         2.000000   
25%               3.000000         2.000000         3.000000   
50%               4.000000         3.000000         3.000000   
75%               5.000000         4.000000         4.000000   
max               7.000000         5.000000        10.000000   

       Discount_offered  Cost_of_the_Product  Weight_in_gms  \
count      10999.000000         10999.000000   10999.000000   
mean          13.373216           210.196836    3634.016729   
std           16.205527            48.063272    1635.377251   
min            1.000000            96.000000    1001.000000   
25%            4.000000           169.000000    1839.500000   
50%            7.000000           214.000000    4149.000000   
75%           10.000000           251.000000    5050.000000   
max           65.000000           310.000000    7846.000000   

       Reached.on.Time_Y.N  
count         10999.000000  
mean              0.596691  
std               0.490584  
min               0.000000  
25%               0.000000  
50%               1.000000  
75%               1.000000  
max               1.000000

Some observations:

1. Coloumn `Customer_care_calls`, `customer_rating`, and `Prior_purchases` the distribution seems to be quite symmetrical (mean and median are not much different)
2. Coloumn `Discount_offered` and `Cost_of_the_Product`  seems to skew to the right (long-right tail)
3. Coloumn `Reached.on.Time_Y.N` boolean/binary value

In [8]:
df[cats].describe()

Mode_of_Shipment Product_importance Gender Warehouse_block
count             10999              10999  10999           10999
unique                3                  3      2               5
top                Ship                low      F               F
freq               7462               5297   5545            3666